diff --git wordpress-orig/wp-includes/functions.php wordpress/wp-includes/functions.php
index 098079577e..2574ba6c32 100644
--- wordpress-orig/wp-includes/functions.php
+++ wordpress/wp-includes/functions.php
@@ -6175,8 +6175,8 @@ function wp_delete_file( $file ) {
  * @return bool True on success, false on failure.
  */
 function wp_delete_file_from_directory( $file, $directory ) {
-	$real_file      = realpath( wp_normalize_path( $file ) );
-	$real_directory = realpath( wp_normalize_path( $directory ) );
+	$real_file      = wp_is_stream( $file ) ? wp_normalize_path( $file ) : realpath( wp_normalize_path( $file ) );
+	$real_directory = wp_is_stream( $directory ) ? wp_normalize_path( $directory ) : realpath( wp_normalize_path( $directory ) );

	if ( false === $real_file || false === $real_directory || strpos( wp_normalize_path( $real_file ), trailingslashit( wp_normalize_path( $real_directory ) ) ) !== 0 ) {
		return false;

